### ISSUE TITLE
Support MLX for Mac

### ISSUE THREAD

**Issue by remichu-ai** [Dan-wanna-M/formatron#24]
Hi, would like to ask if there is plan to support mlx inference engine. 

It is the current fastest engine for mac os and i dont think any of the existing other library that support it for format constraints yet.

**Comment by remichu-ai** [comment]
Can you provide some guidance on what to do, i will try to do a pr. It will be good learning for me as well

**Comment by Dan-wanna-M** [comment]
@remichu-ai You can start by looking at the integrations folder in formatron repositories. You probably also need to take a look at MLC's logits processor API. The basic idea is that we create some utilities/glue code so that formatron updates the state machine by accepting the current token and then restricting the logits for the next token. The details however are dependent on specific inference engines(that's why I write different modules for different inference engines; inherently they are not the same.)
